### DISCUSSION TITLE
Git hash in log file

### DISCUSSION THREAD

**Discussion by bobturneruk** [pykale/pykale#83] [Q&A] [answered]
Hi everyone! Is the purpose of this function simply to help generate unique filenames, or is it important to include the git hash (as a reference to a version of the code) in the log?

https://github.com/pykale/pykale/blob/9a6367a0b017a2d59020eabc918e772bf43d29f3/kale/utils/logger.py#L11

**Comment by haipinglu** [comment, marked as answer]
Here, we just use it for generating unique filenames. It is more of legacy code that we may consider to remove in future since there are better loggers out there that our more recent code has been using.

**Reply by bobturneruk** [reply]
So a random number / code would be fine?

**Reply by bobturneruk** [reply]
(I think there's also a possibility that the current approach of combining git hash and date will not guarantee uniqueness e.g. if the code is run twice in one minute at the same commit.)

**Reply by haipinglu** [reply]
> (I think there's also a possibility that the current approach of combining git hash and date will not guarantee uniqueness e.g. if the code is run twice in one minute at the same commit.)

Yes, that's right so this logger does have some issues. You may simplify it first (e.g. remove the git hash) and later, we may consider to remove it and replace it with other loggers in the three examples. It was only used in our older examples.

**Reply by bobturneruk** [reply]
Also - is this line useful?
https://github.com/pykale/pykale/blob/2209cff8e8b8d4c5f06ec16710051db3d57b7548/kale/utils/logger.py#L37
It creates a file that is overwritten by each run. Perhaps if the output is useful it should have a unique filename that matches the log?

**Reply by haipinglu** [reply]
To be honest, not sure. We took this from the link at the top but did not figure out the exact intention (for all lines) so I think you could modify as you see appropriate. Many thanks.
